test/stream_test.cc

```diff
@@ -83,6 +86,9 @@ int main(int argc, char** argv) {
     CHECK(byte_stream != nullptr);
 
     auto writer = byte_stream->OpenWriter(writer_client);
+    auto failed_writer = byte_stream->OpenWriter(writer_client);
+    CHECK(failed_writer == nullptr);
```

Comment:
For the check, you may also need to consider all places the `OpenReader/OpenWriter` are called, not merely the test case.